specs/gloas/beacon-chain.md

```diff
@@ -413,34 +493,75 @@ def is_parent_block_full(state: BeaconState) -> bool:
 
 ### Misc
 
-#### Modified `get_pending_balance_to_withdraw`
+#### New `get_builder_index`
+
+```python
+def get_builder_index(state: BeaconState, pubkey: BLSPubkey) -> BuilderIndex:
+    builder_pubkeys = [v.pubkey for v in state.builders]
+    assert pubkey in builder_pubkeys
+    return BuilderIndex(builder_pubkeys.index(pubkey))
+```
+
+#### New `get_validator_index`
+
+```python
+def get_validator_index(state: BeaconState, pubkey: BLSPubkey) -> ValidatorIndex:
+    validator_pubkeys = [v.pubkey for v in state.validators]
+    assert pubkey in validator_pubkeys
+    return ValidatorIndex(validator_pubkeys.index(pubkey))
+```
+
+#### New `convert_builder_index_to_validator_index`
+
+```python
+def convert_builder_index_to_validator_index(builder_index: BuilderIndex) -> ValidatorIndex:
+    return ValidatorIndex(builder_index | BUILDER_INDEX_FLAG)
+```
+
+#### New `convert_validator_index_to_builder_index`
+
+```python
+def convert_validator_index_to_builder_index(validator_index: ValidatorIndex) -> BuilderIndex:
+    return BuilderIndex(validator_index & ~BUILDER_INDEX_FLAG)
+```
 
-*Note*: `get_pending_balance_to_withdraw` is modified to account for pending
-builder payments.
+#### New `get_pending_balance_to_withdraw_for_builder`
 
 ```python
-def get_pending_balance_to_withdraw(state: BeaconState, validator_index: ValidatorIndex) -> Gwei:
+def get_pending_balance_to_withdraw_for_builder(
+    state: BeaconState, builder_index: BuilderIndex
+) -> Gwei:
     return (
         sum(
             withdrawal.amount
             for withdrawal in state.pending_partial_withdrawals
-            if withdrawal.validator_index == validator_index
+            if withdrawal.validator_index == convert_builder_index_to_validator_index(builder_index)
```

Comment:
Is iterating over `pending_partial_withdrawals` necessary to compute the builder's withdrawing balance?